woocommerce/woocommerce
# [Product Blocks editor]: enable/disable header Update button depending on product content changes

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This PR checks if the product has changes to save, following the core [isEditedPostDirty()](https://github.com/WordPress/gutenberg/blob/90aa9bd9ae808dca6c37d5bd6c3a09ca00602a26/packages/editor/src/store/selectors.js#L101) way. 

The defines the `disable` boolean const and exposes it in the usePublish() hook. Since the props are passed to the `<PublishButton />`, the UI reflects the changes.





Closes https://github.com/woocommerce/woocommerce/issues/39991



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to edit a Product
2. Confirm the `Update` button shows initially disabled

<img width="754" alt="Screenshot 2023-11-15 at 12 46 14" src="https://github.com/woocommerce/woocommerce/assets/77539/af9b6a3d-95f4-47d1-a7a1-e6e7681545a4">

3. Add some changes to the product, like editing the title, content, etc.
4. Confirm the `Update` button enables

<img width="749" alt="Screenshot 2023-11-15 at 12 49 02" src="https://github.com/woocommerce/woocommerce/assets/77539/d52bd2c3-b99b-4dbb-a9ef-799bfb6274d2">

5. Save the Product

<img width="737" alt="Screenshot 2023-11-15 at 12 49 23" src="https://github.com/woocommerce/woocommerce/assets/77539/e15942ce-2dc0-4cd0-a2fa-a5acc9638144">

6. Confirm the button gets disabled right after saving the product



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
